### PR TITLE
Juj/add metering point link repo for settlement reports

### PR DESCRIPTION
**Pull request by FirestarJes**
# Description

Added a repo to handle request for the Charge Link Periods basis data

## Pull-request quality

- [x] The title adheres to [this guide](https://github.com/Mech0z/GitHubGuidelines)
- [x] Tests are written and executed locally
- [ ] Subsystem tests have been tested (by manually deploying to `dev_002`)

### CODE REVIEW

File: source/dotnet/wholesale-api/CalculationResults/CalculationResults.Application/SettlementReports_v2/SettlementReportRequestHandler.cs [hunk 1]
```diff
@@ -109,6 +129,23 @@ await foreach (var splitFileRequest in SplitFileRequestPerGridAreaAsync(resultWh
         }
     }
 
+    private async IAsyncEnumerable<SettlementReportFileRequestDto> RequestFilesForChargeLinkPeriodsAsync(
+        SettlementReportFileContent fileContent,
+        SettlementReportRequestId requestId,
+        SettlementReportRequestDto reportRequest)
+    {
+        var resultWholesale = new SettlementReportFileRequestDto(
```
Comment (defectiveAi) [inline source/dotnet/wholesale-api/CalculationResults/CalculationResults.Application/SettlementReports_v2/SettlementReportRequestHandler.cs:137]: nitpick: variable name.

File: source/dotnet/wholesale-api/CalculationResults/CalculationResults.Infrastructure/SettlementReports_v2/Statements/SettlementReportChargeLinkPeriodsQueryStatement.cs [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+                FROM 
+                    {_deltaTableOptions.Value.SettlementReportSchemaName}.{_deltaTableOptions.Value.CHARGE_LINK_PERIODS_V1_VIEW_NAME}
+                WHERE 
+                        {SettlementReportChargeLinkPeriodsViewColumns.GridArea} = '{_filter.GridAreaCode}' AND
+                        {SettlementReportChargeLinkPeriodsViewColumns.CalculationType} = '{CalculationTypeMapper.ToDeltaTableValue(_filter.CalculationType)}' AND
+                        {SettlementReportChargeLinkPeriodsViewColumns.FromDate} >= '{_filter.PeriodStart}' AND
+                        {SettlementReportChargeLinkPeriodsViewColumns.ToDate} < '{_filter.PeriodEnd}' AND
+                        {SettlementReportChargeLinkPeriodsViewColumns.CalculationId} = '{_filter.CalculationId}'
+                ORDER BY 
+                       {SettlementReportChargeLinkPeriodsViewColumns.CalculationId} LIMIT {_take} OFFSET {_skip}
```
Comment (defectiveAi) [inline source/dotnet/wholesale-api/CalculationResults/CalculationResults.Infrastructure/SettlementReports_v2/Statements/SettlementReportChargeLinkPeriodsQueryStatement.cs:60]: Remember to use correct orderby

File: source/dotnet/wholesale-api/CalculationResults/CalculationResults.Interfaces/SettlementReports/Model/SettlementReportChargeLinkPeriodsRow.cs [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+using NodaTime;
+
+namespace Energinet.DataHub.Wholesale.CalculationResults.Interfaces.SettlementReports.Model;
+
+public sealed record SettlementReportChargeLinkPeriodsRow(
+    string MeteringPointId,
+    MeteringPointType? MeteringPointType,
+    ChargeType ChargeType,
+    string ChargeOwnerId,
+    string? ChargeCode,
```
Comment (defectiveAi) [inline source/dotnet/wholesale-api/CalculationResults/CalculationResults.Interfaces/SettlementReports/Model/SettlementReportChargeLinkPeriodsRow.cs:27]: ChargeCode is not null according to contract?

File: source/dotnet/wholesale-api/CalculationResults/CalculationResults.Interfaces/SettlementReports/Model/SettlementReportChargeLinkPeriodsRow.cs [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+using Energinet.DataHub.Wholesale.CalculationResults.Interfaces.CalculationResults.Model;
+using Energinet.DataHub.Wholesale.CalculationResults.Interfaces.CalculationResults.Model.WholesaleResults;
+using Energinet.DataHub.Wholesale.Common.Interfaces.Models;
+using NodaTime;
+
+namespace Energinet.DataHub.Wholesale.CalculationResults.Interfaces.SettlementReports.Model;
+
+public sealed record SettlementReportChargeLinkPeriodsRow(
+    string MeteringPointId,
+    MeteringPointType? MeteringPointType,
```
Comment (defectiveAi) [inline source/dotnet/wholesale-api/CalculationResults/CalculationResults.Interfaces/SettlementReports/Model/SettlementReportChargeLinkPeriodsRow.cs:24]: MeteringPointType is not null according to contract.

File: source/dotnet/wholesale-api/CalculationResults/CalculationResults.Interfaces/SettlementReports/ISettlementReportChargeLinkPeriodsQueries.cs [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+
+using Energinet.DataHub.Wholesale.CalculationResults.Interfaces.SettlementReports.Model;
+
+namespace Energinet.DataHub.Wholesale.CalculationResults.Interfaces.SettlementReports;
+
+public interface ISettlementReportChargeLinkPeriodsQueries
+{
+Task<int> CountAsync(SettlementReportChargeLinkPeriodQueryFilter filter);
+
+IAsyncEnumerable<SettlementReportChargeLinkPeriodsRow> GetAsync(SettlementReportChargeLinkPeriodQueryFilter filter, int skip, int take);
```
Comment (defectiveAi) [inline source/dotnet/wholesale-api/CalculationResults/CalculationResults.Interfaces/SettlementReports/ISettlementReportChargeLinkPeriodsQueries.cs:23]: nitpick: Alignment.

File: source/dotnet/wholesale-api/CalculationResults/CalculationResults.Infrastructure/SettlementReports_v2/Statements/SettlementReportChargeLinkPeriodsQueryStatement.cs (hunk 2 above)
Comment (FirestarJes) [reply source/dotnet/wholesale-api/CalculationResults/CalculationResults.Infrastructure/SettlementReports_v2/Statements/SettlementReportChargeLinkPeriodsQueryStatement.cs:61]: Fixed

File: source/dotnet/wholesale-api/CalculationResults/CalculationResults.Interfaces/SettlementReports/Model/SettlementReportChargeLinkPeriodsRow.cs (hunk 3 above)
Comment (FirestarJes) [reply source/dotnet/wholesale-api/CalculationResults/CalculationResults.Interfaces/SettlementReports/Model/SettlementReportChargeLinkPeriodsRow.cs:27]: Fixed

File: source/dotnet/wholesale-api/CalculationResults/CalculationResults.Interfaces/SettlementReports/Model/SettlementReportChargeLinkPeriodsRow.cs (hunk 4 above)
Comment (FirestarJes) [reply source/dotnet/wholesale-api/CalculationResults/CalculationResults.Interfaces/SettlementReports/Model/SettlementReportChargeLinkPeriodsRow.cs:24]: True, but the mapper can return a null value, so would need a sane default value, which can't be provided.

I could also rewrite the mapper function to not be nullable, but that would mess up everywhere else it's used, and creating a new mapping function for this, seems like overkill?

File: source/dotnet/wholesale-api/CalculationResults/CalculationResults.Interfaces/SettlementReports/ISettlementReportChargeLinkPeriodsQueries.cs (hunk 5 above)
Comment (FirestarJes) [reply source/dotnet/wholesale-api/CalculationResults/CalculationResults.Interfaces/SettlementReports/ISettlementReportChargeLinkPeriodsQueries.cs:23]: Fixed

File: source/dotnet/wholesale-api/CalculationResults/CalculationResults.Interfaces/SettlementReports/Model/SettlementReportChargeLinkPeriodsRow.cs (hunk 4 above)
Comment (FirestarJes) [reply source/dotnet/wholesale-api/CalculationResults/CalculationResults.Interfaces/SettlementReports/Model/SettlementReportChargeLinkPeriodsRow.cs]: Added new method

**Review by defectiveAi — APPROVED** [review]

**Review by defectiveAi — DISMISSED** [review]

**Review by defectiveAi — APPROVED** [review]

**Pull request merged**